src/core/SchrodingerEvolution_CrossCorrelation.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * MRCPP, a numerical library based on multiresolution analysis and
+ * the multiwavelet basis which provide low-scaling algorithms as well as
+ * rigorous error control in numerical computations.
+ * Copyright (C) 2021 Stig Rune Jensen, Jonas Juselius, Luca Frediani and contributors.
+ *
+ * This file is part of MRCPP.
+ *
+ * MRCPP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MRCPP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with MRCPP.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * For information on the complete list of contributors to MRCPP, see:
+ * <https://mrcpp.readthedocs.io/>
+ */
+
+
+#include "SchrodingerEvolution_CrossCorrelation.h"
+
+#include <fstream>
+
+#include "MRCPP/config.h"
+#include "MRCPP/constants.h"
+
+#include "utils/Printer.h"
+#include "utils/details.h"
+
+using namespace Eigen;
+
+namespace mrcpp {
+
+
+/** @brief SchrodingerEvolution_CrossCorrelation constructor.
+ *
+ * @param[in] amount: the integer specifying the maximum amount of matrices \f$ C^k \f$
+ *                    to be used in calculations
+ * @param[in] k: the integer specifying the polynomial order
+ * @param[in] t: the integer specifying the scaling basis type
+ *
+ * @details It checks if the order and type are meaningful and then reads matrices from a file.
+ * By default the file has some information about the data stored,
+ * so the first interger to read is describing the size of the documentation text.
+ * 
+ * 
+ */
+SchrodingerEvolution_CrossCorrelation::SchrodingerEvolution_CrossCorrelation(int amount, int k, int t)
+    : type(t), order(k), amount(amount)
+{
+    if (this->order < 1 or this->order > MaxOrder) MSG_ABORT("Invalid cross correlation order: " << this->order);
+    switch (this->type) {
+        case (Interpol):
+            MSG_ERROR("Not implemented yet filter type: " << this->type);
+        case (Legendre):
+            break;
+        default:
+            MSG_ERROR("Unknown filter type: " << this->type);
+    }
+
+    setCCCPath(details::find_filters());
+
+    readCCCBin();
+}
+
+
+void SchrodingerEvolution_CrossCorrelation::setCCCPath(const std::string &lib) {
+    switch (this->type) {
+        case (Interpol):
+            MSG_ERROR("Not implemented yet filter type: " << this->type);
+            break;
+        case (Legendre):
+            this->path = lib + "/Schrodinger_evolution_cross_correlation_coefficients_Legendre_scaling_type.bin";
```

Comment:
This should be done the same way as the usual cross correlation coeffiients. A binary file will most likely not be portable.